### PR TITLE
Team1 - Initial implementation of client

### PR DESCRIPTION
**Pull request by EdvardJSHolen**

### CODE REVIEW

**Comment by neeldug** [comment]
Rebased to avoid merge commit.

File: pkg/infra/.env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MODE=default
+LEVELS=60
+STARTING_HP=1000
+STARTING_ATTACK=20
+STARTING_SHIELD=20
+BASE_STAMINA=2000
+THRESHOLD_PCT=0.6
+AGENT_RANDOM_QUANTITY=0
+AGENT_TEAM1_QUANTITY=100
```
Comment (neeldug) [inline pkg/infra/.env:9]:
```suggestion
AGENT_TEAM1_QUANTITY=100

```

File: pkg/infra/game/agent/team1.go [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+	var outputArray [3]float64
+
+	outputArray[0] = inputArray[0]
+
+	for i := 1; i < 3; i++ {
+		outputArray[i] = outputArray[i-1] + inputArray[i]
+	}
+
+	return outputArray
+}
```
Comment (neeldug) [inline pkg/infra/game/agent/team1.go:108]: Perhaps move to a helpers module?

File: pkg/infra/go.sum [hunk 3]
```diff
@@ -3,6 +3,12 @@ github.com/benbjohnson/immutable v0.4.0/go.mod h1:iAr8OjJGLnLmVUr9MZ/rz4PWUy6Ouc
 github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/galeone/tensorflow/tensorflow/go v0.0.0-20221023090153-6b7fa0680c3e h1:9+2AEFZymTi25FIIcDwuzcOPH04z9+fV6XeLiGORPDI=
+github.com/galeone/tensorflow/tensorflow/go v0.0.0-20221023090153-6b7fa0680c3e/go.mod h1:TelZuq26kz2jysARBwOrTv16629hyUsHmIoj54QqyFo=
+github.com/galeone/tfgo v0.0.0-20221023090852-d89a5c7e31e1 h1:fsU+Je1A2kNu7e1LUAyOUrzXk50/2yP1tFGqKyNkOsI=
+github.com/galeone/tfgo v0.0.0-20221023090852-d89a5c7e31e1/go.mod h1:3YgYBeIX42t83uP27Bd4bSMxTnQhSbxl0pYSkCDB1tc=
+github.com/golang/protobuf v1.5.0/go.mod h1:FsONVRAS9T7sI+LIUmWTfcYkHO4aIWwzhcaSAoJOfIk=
+github.com/google/go-cmp v0.5.5/go.mod h1:v8dTdLbMG2kIc/vJvl+f65V22dbkXbowE6jgT/gNBxE=
```
Comment (neeldug) [inline pkg/infra/go.sum:11]:
```suggestion
```

File: pkg/infra/go.sum [hunk 4]
```diff
@@ -18,6 +24,10 @@ golang.org/x/exp v0.0.0-20220518171630-0b5c67f07fdf h1:oXVg4h2qJDd9htKxb5SCpFBHL
 golang.org/x/exp v0.0.0-20220518171630-0b5c67f07fdf/go.mod h1:yh0Ynu2b5ZUe3MQfp2nM0ecK7wsgouWTDN0FNeJuIys=
 golang.org/x/sys v0.0.0-20220715151400-c0bba94af5f8 h1:0A+M6Uqn+Eje4kHMK80dtF3JCXC4ykBgQG4Fe06QRhQ=
 golang.org/x/sys v0.0.0-20220715151400-c0bba94af5f8/go.mod h1:oPkhp1MJrh7nUepCBck5+mAzfO9JrbApNNgaTdGDITg=
+golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
+google.golang.org/protobuf v1.26.0-rc.1/go.mod h1:jlhhOSvTdKEhbULTjvd4ARK9grFBp09yW+WbY/TyQbw=
+google.golang.org/protobuf v1.28.1 h1:d0NfwRgPtno5B1Wa6L2DAG+KivqkdutMf1UhdNx175w=
+google.golang.org/protobuf v1.28.1/go.mod h1:HV8QOd/L58Z+nl8r43ehVNZIU/HEI6OcFqwMG9pJV4I=
```
Comment (neeldug) [inline pkg/infra/go.sum:30]:
```suggestion
```

File: pkg/infra/main_test.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"flag"
+	"infra/logging"
+	"testing"
+)
+
+func TestMain(m *testing.M) {
```
Comment (neeldug) [inline pkg/infra/main_test.go:9]: This is unused as of now, main testing requires more tests to be provided in separate functions that will use main test body as closures.

**Review by neeldug — CHANGES_REQUESTED** [review]

**Comment by sherwin-dc** [comment]
After attempting to rebase on main there are errors in the following lines, because `view.AgentState()` behaves differently and does not return a pointer anymore. Does anyone know what this should be changed to?

```go
       // Ensure that socialCapital map is initialised
	if len(r.socialCapital) == 0 && view.AgentState().Len() > 1 {
```

```go
		// Populate map with every currently living agent, and calculate socialCapital based on log
		itr := view.AgentState().Iterator()
```

**Comment by neeldug** [comment]
> After attempting to rebase on main there are errors in the following lines, because `view.AgentState()` behaves differently and does not return a pointer anymore. Does anyone know what this should be changed to?
> 
> ```go
>        // Ensure that socialCapital map is initialised
> 	if len(r.socialCapital) == 0 && view.AgentState().Len() > 1 {
> ```
> 
>  ```go
> 		// Populate map with every currently living agent, and calculate socialCapital based on log
> 		itr := view.AgentState().Iterator()
> ```

Similar to what @Razvan00Rusu said if you just add this line and refer to the variable it should work identically:

```go
agentState := v.AgentState()
```

**Comment by sherwin-dc** [comment]
It seems your suggestions have worked. I have fixed this in #73

**Comment by neeldug** [comment]
Closed in favour of #73

**Pull request closed without merging**